Fix `go vet` issue

/area API
/area test-and-release

## What version of Knative?

HEAD

## Expected Behavior

`go vet ./pkg/...` is clean

## Actual Behavior

```
go vet ./pkg/...
# knative.dev/serving/pkg/apis/serving/v1alpha1
pkg/apis/serving/v1alpha1/service_types.go:111:2: struct field DeprecatedGeneration repeats json tag "generation" also at configuration_types.go:77
```

This is blocking us from picking up the Github actions (which currently include a `go vet` check).  The problem is that we inline types with conflicting definitions of `DeprecatedGeneration`.

cc @vagababov @markusthoemmes @dprotaso  for any brilliant ideas on how to surgically address this without a bunch of churn 🤞

--- a/pkg/apis/serving/v1alpha1/route_types.go
+++ b/pkg/apis/serving/v1alpha1/route_types.go
@@ -89,7 +89,7 @@ type RouteSpec struct {
 	// Tracking issue: https://github.com/knative/serving/issues/643
 	//
 	// +optional
-	DeprecatedGeneration int64 `json:"generation,omitempty"`
+	DeprecatedGeneration int64
 
 	// Traffic specifies how to distribute traffic over a collection of Knative Serving Revisions and Configurations.
 	// +optional

